flink-tests/src/test/java/org/apache/flink/test/checkpointing/UnalignedCheckpointCompatibilityITCase.java

```diff
@@ -107,28 +119,55 @@ public void test() throws Exception {
 	}
 
 	private Tuple2<String, Map<String, Object>> runAndTakeSavepoint() throws Exception {
-		JobClient jobClient = submitJobInitially(env(startAligned, 0, emptyMap()));
-		Thread.sleep(FIRST_RUN_EL_COUNT * FIRST_RUN_BACKPRESSURE_MS); // wait for all tasks to run and some backpressure from sink
-		Future<Map<String, Object>> accFuture = jobClient.getAccumulators();
-		Future<String> savepointFuture = jobClient.stopWithSavepoint(false, tempFolder().toURI().toString());
-		return new Tuple2<>(savepointFuture.get(), accFuture.get());
+		return withCluster(new Configuration(), miniCluster -> {
+			JobClient jobClient = submitJobInitially(env(startAligned, 0));
+			waitForAllTaskRunning(miniCluster, jobClient.getJobID(), Deadline.fromNow(Duration.of(30, ChronoUnit.SECONDS)));
+			Thread.sleep(FIRST_RUN_BACKPRESSURE_MS); // wait for some backpressure from sink
```

Comment:
It's generally an anti-pattern in tests to "just sleep for a while". Can't you use the backpressure monitoring to wait until there is backpressure? (or would that take longer?)

Reply:
The original test used sleeping for waiting until all jobs are running (hence this bug). However, whether backpressure build up or not is quite irrelevant for the semantics of the test. It's just testing that some outputs are present. I could probably wait on the output metric. 
However, it feels like LOTS of such building blocks are missing and should probably be at a more general place. (I tried to do that with `waitForAllTaskRunning` but it's easily overlooked. Ideally it should be in MiniCluster).

Reply:
Maybe such utilities should also be made available to the user, for writing their own tests.
There are a lot of very useful utilities everywhere in our tests, but it's difficult to discover them. Ideally we have them categorized and standardized somewhere.